pkg/grpc/grpc_util.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package grpc
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/pkg/client/generated/cstor-volume-grpc/v1alpha1"
+	"github.com/openebs/maya/pkg/grpc/api"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+// CmdSnaphotOptions holds the options for snapshot
+// create command
+type CmdSnaphotOptions struct {
+	VolName  string
+	SnapName string
+}
+
+// Validate validates the flag values
+func (c *CmdSnaphotOptions) Validate(cmd *cobra.Command) error {
+	if c.VolName == "" {
+		return errors.New("--volname is missing. Please specify a unique name")
+	}
+	if c.SnapName == "" {
+		return errors.New("--snapname is missing. Please specify a unique name")
+	}
+
+	return nil
+}
+
+//CreateSnapshot creates snapshots
+func CreateSnapshot(volName, snapName, ip string) (*v1alpha1.VolumeSnapCreateResponse, error) {
```

Comment:
Can we have a struct called `snapshot` that exposes these methods & makes use of `CASSnapshot` as it args or property & `CASSnapshot` as the method's return type?
The signature will be more clear if we try to contract it against an interface.

Reply:
Eventually this struct should implement an Interface

Reply:
Now I am feeling that all this should go into `pkg/client/snapshot`

Above package will be used by `app/command` & `pkg/task/v1alpha1`